pkg/registry/apis/provisioning/repository/git/repository.go

```diff
@@ -106,11 +99,9 @@ func (r *gitRepository) Validate() (list field.ErrorList) {
 		list = append(list, field.Invalid(field.NewPath("spec", t, "branch"), cfg.Branch, "invalid branch name"))
 	}
 
-	// If the repository has workflows, we require a token or encrypted token
-	if len(r.config.Spec.Workflows) > 0 {
-		if cfg.Token == "" && len(cfg.EncryptedToken) == 0 {
-			list = append(list, field.Required(field.NewPath("spec", t, "token"), "a git access token is required"))
-		}
+	// TODO: is this check required?  for public repositories it should not be necessary
```

Comment:
This should only run if workflows are not enabled (read-only repositories). It's only necessary when you want to write.